From 3a6996ea837f213beec32703897551587a6727a3 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Mon, 03 Oct 2011 15:50:53 +0000
Subject: appchooser: make the "Find applications online" follow search state

Set the button as insensitive while searching, so that it can't be
accidentally clicked twice.

https://bugzilla.gnome.org/show_bug.cgi?id=651818
---
diff --git a/gtk/gtkappchooserdialog.c b/gtk/gtkappchooserdialog.c
index ac1f0af..8d03cc0 100644
--- a/gtk/gtkappchooserdialog.c
+++ b/gtk/gtkappchooserdialog.c
@@ -137,6 +137,7 @@ search_for_mimetype_ready_cb (GObject      *source,
     }
   else
     {
+      gtk_widget_set_sensitive (self->priv->online_button, TRUE);
       gtk_app_chooser_refresh (GTK_APP_CHOOSER (self->priv->app_chooser_widget));
     }
 
@@ -152,6 +153,8 @@ online_button_clicked_cb (GtkButton *b,
   GtkAppChooserDialog *self = user_data;
 
   self->priv->online_cancellable = g_cancellable_new ();
+  gtk_widget_set_sensitive (self->priv->online_button, FALSE);
+
   _gtk_app_chooser_online_search_for_mimetype_async (self->priv->online,
 						     self->priv->content_type,
 						     GTK_WINDOW (self),
--
cgit v0.9.0.2

